### ISSUE TITLE
server hangs if esi data provider doesn't disconnect the fetch context

### ISSUE THREAD

**Issue by tripodsan** [adobe/helix-simulator#489]
how to reproduce
- ensure that helix-cli uses the dev simulator:
```console
$ cd helix-simulator
$ npm link
$ cd ../helix-cli
$ npm link @adobe/helix-simulator
```
- make the data provider reuse the fetch context:
```diff
diff --git a/src/utils.js b/src/utils.js
index 23a9342..a5aee28 100644
--- a/src/utils.js
+++ b/src/utils.js
@@ -345,7 +345,7 @@ const utils = {
       const { url } = options;
       delete options.url;
       // need extra fetch context, otherwise it causes helix-fetch to hang... !?!
-      const context = createFetchContext();
+      const context = fetchContext; //createFetchContext();
       try {
         const { fetch } = context;
         const ret = await fetch(url, {
@@ -360,7 +360,7 @@ const utils = {
           body,
         };
       } finally {
-        context.disconnectAll();
+        // context.disconnectAll();
       }
     }
     return { toFullyQualifiedURL, get };
```
- start the project and request a page 2 times. the 2nd time it will hang fetching the `head.md`

**Comment by stefan-guggisberg** [comment]
It's hard to debug using the given instructions. There's just  too much noise.

The `helix-simulator` tests all succeed when using the same fetch context for proxy requests and ESI includes. It would be good to have a simple test in `helix-simulator` which reproduces the issue of hanging requests.

**Comment by stefan-guggisberg** [comment]
fixed in [v5.1.22](https://github.com/adobe/helix-simulator/releases/tag/v5.1.22) / `helix-cli v13.1.6`
